### ISSUE TITLE
Filter for root properties

### ISSUE THREAD

**Issue by StefanBelo** [fsprojects/FSharp.Data.GraphQL#296]
I am having problems to create filter for root property, something like that:

```
query {
    markets (filter : { totalMatched_greater_than : 1000.0 }) {
        id
        name
        totalMatched
        status
        requestTime
        updateTime
        selections (filter : { totalMatched_greater_than : 1000.0 }) {
            id
            name
            lastPriceTraded
            totalMatched
            status
        }
    }
}
```

In the code base there is actually no example of such usage. Maybe I am wrong because I just read something about graphQL and trying to implement that in my use case scenario.

**Comment by festim-deari** [comment]
@StefanBelo were you able to find a solution?

**Comment by xperiandri** [comment]
What happens when you do that?

**Comment by xperiandri** [comment]
Does it work for nested fields?

**Comment by festim-deari** [comment]
Hi @xperiandri, sorry for replying late.

This is my middlewares definition:
` let middlewares = 
        [
          Define.ObjectListFilterMiddleware<Root.Root, Label>(true)]`
          
 Root.Root is the ObjectType for the QueryRoot. It has a field called labels of type Label list.
 
 When I try to filter in the query as follows:
 `
query {
	labels  (filter: {annotation_sw: "Cu"}) {
		annotation
		id
	}
}
`

It throws the error :
 "Argument field or value named 'filter' can not be coerced. It does not match a valid literal representation for the type."

**Comment by festim-deari** [comment]
> Does it work for nested fields?

Yes it does in my case. I tried it for a nested field which is also a list, and it works perfectly.
